os/board/rtl8730e/src/component/wifi/inic/inic_ipc_host_api.c

```diff
@@ -323,15 +334,36 @@ static void inic_ipc_print_int_hdl(VOID *Data, u32 IrqStatus, u32 ChanNum)
 	(void) IrqStatus;
 	(void) ChanNum;
 
+	static km4log_msg_t message_event = { 0 };
+
+	/* receive a log message over IPC */
 	PIPC_MSG_STRUCT ipc_recv_msg = (PIPC_MSG_STRUCT)ipc_get_message(IPC_NP_TO_AP, IPC_N2A_NP_LOG_CHN);
 	char *tmp_buffer = (char *)ipc_recv_msg->msg;
 	DCache_Invalidate((u32)tmp_buffer, ipc_recv_msg->msg_len);
-	
-	/* Print out buffer */
-	lldbg_noarg("%s",tmp_buffer);
 
+	/* pick a buffer to fill. for rapid processing, assume that the higher priority task has already consumed the buffer, otherwise overwrite it */
+	strncpy((char *)g_inic_ipc_logging_buf[g_inic_ipc_logging_buf_ctr], tmp_buffer, ipc_recv_msg->msg_len);
+
+	/* fill message struct */
+	message_event.buffer = (void *)g_inic_ipc_logging_buf[g_inic_ipc_logging_buf_ctr];
+	message_event.buffer_len = CONFIG_KM4_MAX_BUFFER_SIZE;
+
+	/* use mq_send via osif api directly in ISR instead of semaphore-based */
+	if (g_km4_log_queue == NULL || (!osif_msg_send(g_km4_log_queue, &message_event, 0))) {
```

Comment:
Is this mq_send from ISR to log thread?
ISR and log thread both are in BSP. And ISR does not have other context.
How about directly copying logs into the buffer?
@kishore-sn @gSahitya-samsung @ewoodev @jeongarmy How do you think?

Reply:
@lhenry-realtek @zhongnuo-tang Sorry, please ignore my comment above. Let's keep current.